client/src/components/privateRoutes.tsx

```diff
@@ -0,0 +1,20 @@
+import { useContext } from 'react';
+import { Navigate, useLocation } from 'react-router-dom';
+import { UserContext } from '../context';
+import { PrivateType, UserContextTypeWithDispatch } from '../interfaces';
+
+function ProtectedRoute({ children, roles }:PrivateType) {
```

Comment:
```suggestion
function ProtectedRoute({ children, roles }:PrivateType) {
```
Not sure what it's `roles` insteadof `role`